protocol/strict-derivation.md

```diff
@@ -0,0 +1,355 @@
+# Purpose
+
+Holocene aims to introduce a set of changes to derivation that improve worst-case scenarios for
+Fault Proofs and Interop. The goal of this design doc is to align on the changes to derivation that
+will be part of the upcoming Holocene hardfork, and in particular to solve open questions to
+finalize the specs.
+
+The Holocene specs for these derivation changes are currently being written in [specs/#357](https://github.com/ethereum-optimism/specs/pull/357).
+
+# Summary
+
+The Holocene hardfork introduces several changes to batch derivation:
+
+- _Partial Span Batch Validity_ determines the validity of singular batches within a span batch
+individually, only invalidating the remaining span batch upon the first invalid batch.
+- _Steady Batch Derivation_ derives invalid singular batches immediately as deposit-only
+blocks.
+- _Strict Batch Ordering_ requires batches within and across channels to be strictly ordered.
+
+The design space and some proposed solutions will be discussed, together with practical implications
+for batcher implementations that have to satisfy the stricter rules.
+
+# Partial Span Batch Validity
+
+## Problem Statement + Context
```

Comment:
One aspect of the problem that is worth mentioning, although solved by the same partial-validity idea, is that interop fault-proofs require a block to be processed optimistically, and then the proof might later abort on cross-L2 interop-dependencies. After having aborted, the alternative chain continuation should be as straight-forward and minimal as possible, to avoid looping interop dependency checks. Falling back to a deposit-only block, when a batch is invalid, would serve this well.